fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/FenixOverlay.kt

```diff
@@ -9,21 +9,29 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.rememberCoroutineScope
 import androidx.navigation.compose.rememberNavController
+import mozilla.components.browser.state.store.BrowserStore
 import mozilla.components.lib.state.ext.observeAsState
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
 import org.mozilla.fenix.debugsettings.navigation.DebugDrawerRoute
 import org.mozilla.fenix.debugsettings.store.DebugDrawerAction
 import org.mozilla.fenix.debugsettings.store.DebugDrawerNavigationMiddleware
 import org.mozilla.fenix.debugsettings.store.DebugDrawerStore
 import org.mozilla.fenix.debugsettings.store.DrawerStatus
+import org.mozilla.fenix.debugsettings.tabs.TabTools
 import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.theme.Theme
 
 /**
  * Overlay for presenting Fenix-wide debugging content.
+ *
+ * @param browserStore [BrowserStore] used to access tab data for [TabTools].
+ * @param inactiveTabsEnabled Whether the inactive tabs feature is enabled.
  */
 @Composable
-fun FenixOverlay() {
+fun FenixOverlay(
+    browserStore: BrowserStore,
+    inactiveTabsEnabled: Boolean,
+) {
```

Comment:
Thinking aloud as I double check the diff, we could alternatively/optionally keep these outside of `HomeActivity` by doing:

```
fun FenixOverlay() {
   FenixOverlayContent(
       browserStore = components.core.store,
       inactiveTabsEnabled = components.settings.inactiveTabsEnabled,
   )
}

private fun FenixOverlayContent(
    browserStore: BrowserStore,
    inactiveTabsEnabled: Boolean,
) {
   // the existing set up and observation calls
}

private fun FenixOverlayPreview() {
    FenixOverlayContent(
        browserStore = BrowserStore(),
        inactiveTabsEnabled = true,
    )
}
}
```

The content abstraction would then keep previews working, since calls to `components` breaks those.

I don't feel strongly toward this because of my aversion to using the `components` singleton, but this would help keep `HomeActivity` from blowing up as more features are added to the debug drawer.

Reply:
This seems like it might be a problem of clear dependencies, vs the `components` service locator pattern specifically. I don't think hiding the dependencies in `FenixOverlay` using the `Content` Composable is the way to go. `HomeActivity` may eventually become a bloated depency injection machine, but I think if that were its only job that would be closer to optimal than today anyway.